File: app/src/main/java/ru/garretech/garred/doramatv/database/AppDatabase.java
package ru.garretech.garred.doramatv.database;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.migration.Migration;
import android.content.Context;

import ru.garretech.garred.doramatv.model.Favorites;
import ru.garretech.garred.doramatv.model.Movie;

@Database(entities = {Movie.class,Favorites.class},version = 2, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static final String DATABASE_NAME = "app_database";
    private static AppDatabase INSTANCE = null;


    public static AppDatabase getInstance(Context context) {
        if (INSTANCE == null) {
         synchronized (AppDatabase.class) {

             INSTANCE = Room.databaseBuilder(context,AppDatabase.class,DATABASE_NAME)
                     .fallbackToDestructiveMigration()
                     .build();
         }
        }
        return INSTANCE;
    }

    public abstract MovieDAO movieDAO();
    public abstract FavoritesDAO favoritesDAO();


    public static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(final SupportSQLiteDatabase database) {
            //database.execSQL("ALTER TABLE favorites DROP TO birthday INTEGER DEFAULT 0 NOT NULL");
        }
    };
}

File: app/src/main/java/ru/garretech/garred/doramatv/database/AppDatabase.kt
package ru.garretech.garred.doramatv.database

import androidx.sqlite.db.SupportSQLiteDatabase
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.migration.Migration
import android.content.Context

import ru.garretech.garred.doramatv.model.Favorites
import ru.garretech.garred.doramatv.model.Movie

@Database(entities = [Movie::class, Favorites::class], version = 3, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun movieDAO(): MovieDAO
    abstract fun favoritesDAO(): FavoritesDAO

    companion object {

        private val DATABASE_NAME = "app_database"
        private var INSTANCE: AppDatabase? = null


        fun getInstance(context: Context): AppDatabase? {
            if (INSTANCE == null) {
                synchronized(AppDatabase::class.java) {

                    INSTANCE = Room.databaseBuilder(context, AppDatabase::class.java, DATABASE_NAME)
                            .fallbackToDestructiveMigration()
                            .build()
                }
            }
            return INSTANCE
        }


        val MIGRATION_2_3: Migration = object : Migration(2, 3) {
            override fun migrate(database: SupportSQLiteDatabase) {
                //database.execSQL("ALTER TABLE favorites DROP TO birthday INTEGER DEFAULT 0 NOT NULL");
            }
        }
    }
}
